src/app/components/import-data/import-data.component.ts

```
@@ -0,0 +1,168 @@
+    @Input() comp: ConfigurableWidget;
+    @Input() sideNavRight: MatSidenav;
+
+    private dataset: Dataset;
+    private messenger: eventing.Messenger;
+    public optionCollection: WidgetOptionCollection;
+    public isFileSelected: boolean;
+    public readonly dashboardState: DashboardState;
+
+    public csvParseError: string;
+    public warningMessage: string;
+
+    constructor(
+        dashboardService: DashboardService,
+        protected connectionService: InjectableConnectionService,
+        @Inject(Object) private papa: any
+    ) {
+        this.messenger = new eventing.Messenger();
+        this.dashboardState = dashboardService.state;
+        let stateSourceSubscription = dashboardService.stateSource.subscribe((dashboardState) => {
+            this.dataset = dashboardState.asDataset();
+            this.optionCollection = new WidgetOptionCollection(this.dataset);
+        });
+
+        stateSourceSubscription.unsubscribe();
+    }
+
+    ngOnDestroy() {
+        this.messenger.unsubscribeAll();
+    }
+
+    public onDatastoreChanged() {
+        this.optionCollection.updateDatabases(this.dataset);
+        this.warningMessage = null;
+    }
+
+    public onDatabaseChanged() {
+        this.optionCollection.updateTables(this.dataset);
+        this.warningMessage = null;
+    }
+
+    public get selectedFile() {
+        return this.isFileSelected ? this.inputFile.nativeElement.files[0].name : '';
+    }
+
+    public onFileSelected() {
+        this.isFileSelected = true;
+    }
+
+    public onCancel() {
+        this.sideNavRight.close();
+    }
+
+    public onImportClick() {
+        let file = this.inputFile.nativeElement.files[0];
+        this.papa.parse(file, { header: true,
+            complete: this.import.bind(this) });
+    }
+
+    import(result: any) {
+        if (result.errors.length > 0) {
+            this.csvParseError = `CSV parsing error at line ${result.errors[0].row}: ${result.errors[0].message}`;
```

@aamduka Can we show the messages from ALL the `result.errors`, and not just the first one?  I get confused because I think my file only contains one invalid row when it actually contains multiple (since the other errors are hidden).